Kernel/InterpretedLiteralEvaluator.cpp

```diff
@@ -1172,6 +1172,7 @@ bool InterpretedLiteralEvaluator::balance(Literal* lit,Literal*& resLit,Stack<Li
     
     // find which arg of t2 is the non_constant bit, this is what we are unwrapping 
     TermList to_unwrap;
+    to_unwrap.makeEmpty();
```

Comment:
This actually looks like a bug. The default `TermList` constructor does _nothing_, but we need `to_unwrap` to be `isEmpty()` on first access. Possibly this "worked" by chance.

Reply:
Should we maybe quickly ask @joe-hauns  about it?

Reply:
Yes looks like a bug. The code is not by me originally, so I can't say whether I overlooked something but it looks like you're right! 

What do you think of making the default constructor of `TermList` create an empty term instead of creating an uninitalized one? For me this has been a gotcha quite a few times.

Reply:
I think that's sensible. Very likely this doesn't cause much performance overhead; I'll put it on TODO for later.